### PR TITLE
Chore: Upgrade deps and scripts to work with node 10

### PR DESCRIPTION
**Pull request by jstoffan**
I also added an `.npmrc` file and package.json `engines` declaration to replace the current version-checking script.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

**Comment by jstoffan** [comment]
Good catch, @jeremypress. I ran the script and didn't see any output. It looks like it's checking the minimum version, which shouldn't conflict with upgrading to node 10. Seems like everything's okay.

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.org/
```
Comment (priyajeet) [inline .npmrc]: weird that this was missing before :O

Comment (priyajeet) [reply .npmrc]: in other projects we have both yarnrc and npmrc with this

**Pull request merged**